src/overlays/actors/ovl_Dm_Zl/z_dm_zl.c

```diff
@@ -30,27 +30,243 @@ const ActorInit Dm_Zl_InitVars = {
+    gDmZl4EyeWideTex,
+    gDmZl4EyeHalf1Tex,
+    gDmZl4EyeOpenLookingLeftTex,
+    gDmZl4EyeOpenLookingRightTex,
+    NULL,
+    NULL,
+    NULL,
+};
+
+void DmZl_ChangeAnimation(SkelAnime* skelAnime, AnimationInfo animation[], u16 index) {
+    f32 endFrame;
+
+    animation += index;
+    endFrame = (animation->frameCount < 0.0f) ? Animation_GetLastFrame(animation->animation) : animation->frameCount;
+
+    Animation_Change(skelAnime, animation->animation, animation->playSpeed, animation->startFrame, endFrame,
+                     animation->mode, animation->morphFrames);
+}
+
+void DmZl_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    DmZl* this = THIS;
+
+    this->animationIndex = 0;
+    this->unused2BA = 0;
+    this->actor.targetArrowOffset = 1000.0f;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 24.0f);
+    // NULL jointTable, NULL morphTable, limbcount == 0 ? surprised this works
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gZl4Skeleton, NULL, NULL, NULL, 0);
+    DmZl_ChangeAnimation(&this->skelAnime, &sAnimations[this->animationIndex], 0);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = DmZl_DoNothing;
+}
+
+void DmZl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DmZl* this = THIS;
+}
+
+void DmZl_DoNothing(DmZl* this, GlobalContext* globalCtx) {
+}
+
+void DmZl_UpdateCutscene(DmZl* this, GlobalContext* globalCtx) {
+    s32 actionIndex;
+
+    if (Cutscene_CheckActorAction(globalCtx, 0x66) != 0) {
+        actionIndex = Cutscene_GetActorActionIndex(globalCtx, 0x66);
+        if (globalCtx->csCtx.frames == globalCtx->csCtx.actorActions[actionIndex]->startFrame) {
+            s16 nextAnimationIndex = ZELDA_ANIM_FACING_AWAY;
+
+            switch (globalCtx->csCtx.actorActions[actionIndex]->action) {
+                case 2:
+                    nextAnimationIndex = ZELDA_ANIM_TURNING_TOWARD_PLAYER;
+                    break;
+                case 3:
+                    nextAnimationIndex = ZELDA_ANIM_GIVING_OCARINA_START;
+                    break;
+                case 4:
+                    nextAnimationIndex = ZELDA_ANIM_PLAYING_OCARINA_START;
+                    break;
+                case 1:
+                default:
+                    break;
+            }
+
+            if (nextAnimationIndex != this->animationIndex) {
+                this->animationIndex = nextAnimationIndex;
+                DmZl_ChangeAnimation(&this->skelAnime, &sAnimations[this->animationIndex], ZELDA_ANIM_FACING_AWAY);
+            }
+        }
+
+        Cutscene_ActorTranslateAndYaw(&this->actor, globalCtx, actionIndex);
+    }
+
+    if ((Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) != 0) &&
+        (((actionIndex = this->animationIndex) == 1) || (actionIndex == 3) || (actionIndex == 5))) {
+        this->animationIndex++;
+        DmZl_ChangeAnimation(&this->skelAnime, &sAnimations[this->animationIndex], ZELDA_ANIM_FACING_AWAY);
```

Comment:
Is it possible to break the if condition up so that the `actionIndex` assignment doesn't occur in the if?